File: c/tree/invert-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* invertTree(struct TreeNode* root) {
    if(!root)
        return NULL;

    struct TreeNode *node = invertTree(root->left);
    root->left = invertTree(root->right);
    root->right = node;

    return root;
}

File: python/tree/invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def invertTree(self, root):
        if not root:
            return None

        node = self.invertTree(root.left)
        root.left = self.invertTree(root.right)
        root.right = node

        return root
